gateway/api_loader.go

```diff
@@ -121,10 +121,6 @@ func processSpec(spec *APISpec, apisByListen map[string]int,
 		"prefix": "coprocess",
 	})
 
-	if strings.Contains(spec.Proxy.TargetURL, "h2c://") {
```

Comment:
This check moved to reverse proxy code https://github.com/TykTechnologies/tyk/pull/3496/files#diff-e6e07722257f7e41691e471185ad6d84fd56dc9e5459526ea32e9a5e8fa1a01bR1052

Otherwise, inside reverse proxy code while creating the transport we will not know that upstream is h2c.
Plus it does not work with load balancing, or service discovery, so this issue automatically got solved too.